src/Field.tsx

```diff
@@ -171,11 +169,18 @@ class FieldInner<Values = {}, Props = {}> extends React.Component<
     }
 
     if (typeof component === 'string') {
-      const { innerRef, ...rest } = props;
+      const { innerRef, type, ...restProps } = props;
+      const { value, ...restField } = field;
+
       return React.createElement(component as any, {
+        type,
         ref: innerRef,
-        ...field,
-        ...rest,
+        ...restField,
+        ...restProps,
+        ...{
+          [type === 'radio' || type === 'checkbox' ? 'checked' : 'value']:
+            type === 'radio' ? props.value === value : value,
```

Comment:
if `props.value` is an array (multiple checkboxes), we should check if `props.value.includes(value)`. Also since this exact comparison is used twice, I think it should be refactored like `{ ...mapFieldValue(type, value, props.value) }`: 

```
// might exist somewhere already, further reading [https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-8.html](on TypeScript website)
type RemoveArray<ValueType> = ValueType extends (infer ValueTypeWithoutArray)[] ? ValueTypeWithoutArray : ValueType;

export function mapFieldValue<ValueType = any>(type: string, value: RemoveArray<ValueType>, currentValue: ValueType) {
    const key = ['radio', 'checkbox'].includes(type) ? 'checked' : 'value';

    return {
        [key]: type === 'radio'
            ? isArray(props.value) 
                ? props.value.includes(value) 
                : props.value === value 
            : value,
    }
}
```